src/crate/qa/tests.py

```diff
@@ -195,6 +195,7 @@ def setUp(self):
 
         def new_node(version, settings={}):
             crate_dir = get_crate(version)
+            version_tuple = _extract_version(crate_dir)
```

Comment:
what's the difference between `version_tuple` and `v` below? seems like we're doing similar work twice?

Reply:
unfortunately the `v` cuts off the hotfix digit, so 3.1.2 for example becomes 3.1
I can use the v though and add the `.0-SNAPSHOT` to the regex since only the `.0` is the snapshot version, what do you thing?

Reply:
I'd change `version_from_dir` to operate on the `version_tuple`. So you've

```python
version_tuple = _extract_version(crate_dir)
v = version_tuple_to_strict_version(version_tuple)
```